sdk_commons/sdk_commons/config.py

```diff
@@ -0,0 +1,103 @@
+import abc
+from enum import Enum
+
+
+class SDKConfig(abc.ABC):
+    """
+    This is the abstract common interface that every venues
+    are expended to implement. Each method will be invoked by
+    API providing all listed parameters but each venue is free
+    to only include needed parameters in their own concrete
+    implementation. All other parameters will be consumed by
+    *args and **kwargs that are always expected to be included
+    A template venue is also provided as example for a concrete
+    implementation.
+    """
+
+    @property
+    @abc.abstractmethod
+    def authorization_pages(self):
+        """
+        Set this property with a class containing
+        mainnet and testnet properties
+        """
+        pass
+
+    @abc.abstractmethod
+    def venue_chains(self) -> Enum:
+        """
+        Return an Enum listing all supported chains
+        """
+        pass
+
+    @abc.abstractmethod
+    def create_offer(
+        self,
+        address: str,
```

Comment:
what address is that? The swap_contract? I would suggest renaming that to better represent the variable meaning

Reply:
swap address could be too specific, for example Opyn refers to the contract as "Settlement Contract" and not "Swap Contract".
If we want to make the parameter more explicit we could rename as `contract_address`, but maybe that now that we moved these parameters on the constructor it could be not strictly needed. Wdyt?

Reply:
Agree with a more general name. `contract_address` looks good 😃  
The problem with `address` is that bidding tokens, wallets are also addresses.

Not sure if `contract_address` should be in the constructor as it is currently a parameter on RFQ creation. Normally, we use a single contract, but there was a recent case for mult-sign where we used different contracts.

Reply:
Wanna consider this?
https://github.com/tradeparadigm/sdks/runs/7494774772?check_suite_focus=true

Reply:
If not included in the constructor then we will have to re-add it to every call.
The default use of contract_address is to build the ContractConfig and since it is common to every common it makes sense to keep it in the constructor. Any other use or override can be managed as an additional parameter on the specific functions requiring it

```
def specific_function_optionally_requiring_contract_override(self, ......, contract_address: Optional[str] = None):
    config = ContractConfig(
        address=contract_address or self.contract_address,
        ...
```

Reply:
> If not included in the constructor then we will have to re-add it to every call.

Yesterday I started to draft the fixes on API side and I changed my mind. The problem is that having a constructor on the SDK interface make things more complex on API side, while repeat the needed parameters on each method lets a smoother integration.

For the first implementation I will drop the constructor. We can consider it an improvement to be faced on a second step, but much more changes will be needed on the API then